Jenkinsfile

```diff
@@ -0,0 +1,72 @@
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-support-notifications-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/support-notifications/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-support-scheduler-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/support-scheduler/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-sys-mgmt-agent-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/sys-mgmt-agent/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-security-proxy-setup-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/security-proxy-setup/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-security-secretstore-setup-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/security-secretstore-setup/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-edgex-ui-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-ui-go/master/Dockerfile'],
+    [scratch: false, image: 'nexus3.edgexfoundry.org:10004/docker-security-bootstrapper-go:master', dockerfile: 'https://raw.githubusercontent.com/edgexfoundry/edgex-go/master/cmd/security-bootstrapper/Dockerfile']
+]
+pipeline {
+    agent {
+        label 'centos7-docker-4c-2g'
+    }
+    options {
+        timestamps()
+    }
+    parameters {
+        string(
+            name: 'EmailRecipients',
+            defaultValue: 'security-issues@lists.edgexfoundry.org',
+            description: 'The email recipients for Synk HTML report')
+    }
+    triggers {
+        cron '@weekly'
+    }
+    stages {
+        stage('Run Snyk scan') {
+            when {
+                anyOf {
+                    triggeredBy 'UserIdCause'
+                    triggeredBy 'TimerTrigger'
+                }
+            }
+            steps {
+                script {
+                    nexusImages.each {
+                        try{
+                            if (!it.scratch){
+                                sh 'rm -rf Dockerfile'
+                                sh "curl -s ${it.dockerfile} > Dockerfile"
+                                edgeXSnyk(
+                                    command: 'test',
+                                    dockerImage: it.image,
+                                    dockerFile: 'Dockerfile',
+                                    severity: 'high',
+                                    sendEmail: true,
+                                    emailTo: params.EmailRecipients,
+                                    htmlReport: true
+                                )
+                            }
+                        }catch (Exception ex){
+                            println "[ERROR]: ${ex}"
```

Comment:
Will the try/catch cause the build to succeed when errors happen? Should we let the job fail when errors happen?

Reply:
If there are vulnerabilities in an image, the shared library `edgeXSnyk` will mark the build as unstable. Other cases it wouldn't. Do you think we should mark it as failure in case of unexpected errors ?

Reply:
I am not sure the best approach for errors. I might be frustrating scanning 20 images and failing on the 3rd one and the job not completing. Lets try ignoring the errors at the moment. We will just need to keep an eye on this job initially.